Proofread, reorganized, and added to all the "Python on Windows" guides (#794)
discord-python/site: pydis_site/apps/content/resources/guides/python-guides/windows/putting-python-on-path.md

The [python.org Python installer](https://www.python.org/downloads/) provides an option to add Python to your Path
automatically, although it is not checked by default so many people miss it:

[![Add Python to path installer checkbox.](/static/images/content/python-on-windows/installer_path_checkbox.png)](/static/images/content/python-on-windows/installer_path_checkbox.png)

But don't worry, you can have the installer add it after the fact, or add it yourself! Continue on just below to the
[safe method](#safe-method-let-the-installer-add-python-to-path) to have the installer put Python on Path for you. Or,
if you want to learn a handy Windows programming skill, skip to the [advanced
method](#advanced-method-manually-edit-the-path) that explains how to edit the Path manually.

## Safe method: Let the installer add Python to Path

The easiest and safest way to add Python to Path is by rerunning the installer. For this you will need the same
installer you used to install Python, which you can download again if needed from
[python.org](https://www.python.org/downloads/windows/), or it may still be in your Downloads folder (it's a file named

> something like `python-3.11.0-amd64.exe`).

Reviewer: Can you just access it through "apps and features -> python -> modify" in settings.

Author: You can but, as I discovered, you'll still need the installer around to finally make the changes. This is mentioned in parentheses just before the "Verifying your changes" section.

I figured, since they need it anyway, better to just run the installer than look for python on Apps & features.

Reviewer: Makes sense

[here](../installing-and-using-python/#the-py-launcher)), and then after `>>>` in the Python REPL, run

```py
import sys; print(sys.executable); exit();
```

to see that the exact executable that is currently running Python is what you expect:

[![Checking Python executable.](/static/images/content/python-on-windows/testing_path_worked_3.png)](/static/images/content/python-on-windows/testing_path_worked_3.png)

Of course your username will probably not be "r", and your executable path may differ from
`C:\Users\<user>\AppData\Local\Programs\Python\Python311` depending on where you chose to install Python and what
version you have. The last folder will normally be `Python310` for Python 3.10, `Python39` for Python 3.9, etc. But
those differences don't matter if `python` and `pip` and other things like `pyinstaller` (if you've [pip
installed](https://pypi.org/project/pyinstaller/) it) now work for you!

Hopefully things are indeed working, however, it is possible that, due to having multiple Python versions, or other
mixups, the commands are still not behaving how you expect. If so, read on to learn how to manually edit the Path. (If

> you already did that, open a help channel on the [Python Discord server](https://discord.com/invite/python) and someone

Reviewer: ```suggestion
you already did that, open a help channel on the [Python Discord server](https://discord.com/invite/python) explaining everything you've tried so far, and someone
```

will hopefully be able to help.)

## Advanced Method: Manually edit the Path

### About the Path

> As mentioned above, the Path (often called "PATH", though it shows up as "Path") is what Windows uses to locate

Reviewer: I'd used PATH (caps) previously throughout to clearly distinguish between it and a file/folder path, but I don't really mind.

Author: Yeah, I only used "Path" (and usually "the Path") since in the actual environment variables that's the case it shows up as, though a few times I use "PATH" since the python installer calls it that.

I wish the env var was uppercase PATH since then it is more distict and matches how it's usually talked about online.

But, as explained [here](https://superuser.com/a/878382), when actually used, the User Path is appended to the System
Path, forming one long list.

So, how it works is whenever you type a command like `python` into a terminal, the terminal looks for a matching
runnable file like `python.exe` ([or `python.bat`](https://en.wikipedia.org/wiki/Batch_file)) in each of the folders
listed in the Path, starting with the System Path (top to bottom), and then the User Path (top to bottom). It stops
searching on the first one it finds and runs that program with the command line arguments you may have given it, hence
(hopefully) running Python from an installed `python.exe`.

If it can't find a runnable file that matches the command name, it spits out an error like:

```text
foobar: The term 'foobar' is not recognized as the name of a cmdlet, function, script file, or operable program.
```

Or, in the case of not finding `python`, it may [open up the Microsoft
store](../microsoft-store#confusing-app-execution-alias-behaviour).

Again, the Path lookup order is:

> System Path (top to bottom) &rarr; User Path (top to bottom) &rarr; Command not found error

### Adding Python to your Path

> Here the common case of adding Python and the Python Scripts directory to the Windows Path is detailed. Though knowing

Reviewer: ```suggestion
Here only the common case of adding Python and the Python Scripts directory to the Windows Path is detailed. However, knowing
```

    like `pip.exe` and `pyinstaller.exe` (if you have it installed).

    [![Finding exes in folder.](/static/images/content/python-on-windows/finding_exe_3.png)](/static/images/content/python-on-windows/finding_exe_3.png)

    These are the executables we want the command line to be able to find via the Windows Path.

3.  Now, type "environment variables" in the Start menu or Start search box and open the "Edit the system environment
    variables" option.

    [![Step 3](/static/images/content/python-on-windows/edit_path_1.png)](/static/images/content/python-on-windows/edit_path_1.png)

4.  Hit the "Environment Variables..." button.

    [![Step 4](/static/images/content/python-on-windows/edit_path_2.png)](/static/images/content/python-on-windows/edit_path_2.png)

5.  Then in a new window you should see two "Path" variables, one under "User variables for &lt;user&gt;" and one under
    "System variables". Select one of them (most likely the User Path) and hit the "Edit..." button underneath it.

    [![Step 5](/static/images/content/python-on-windows/edit_path_3.png)](/static/images/content/python-on-windows/edit_path_3.png)

>     I suggest using the User Path (what the screenshots show) for the default installation of Python. In general, only

Reviewer: ```suggestion
    You should use the User Path (what the screenshots show) for the default installation of Python. In general, only
```
